skrub/_table_vectorizer.py

```diff
@@ -293,40 +243,35 @@ class TableVectorizer(TransformerMixin, BaseEstimator):
-        If set to `True`, will try to convert each column to the best possible
-        data type (dtype).
-
-    impute_missing : {'auto', 'force', 'skip'}, default='auto'
-        When to impute missing values in categorical (textual) columns.
-        'auto' will impute missing values if it is considered appropriate
-        (we are using an encoder that does not support missing values and/or
-        specific versions of pandas, numpy and scikit-learn).
-        'force' will impute missing values in all categorical columns.
-        'skip' will not impute at all.
-        When imputed, missing values are replaced by the string 'missing'
-        before being encoded.
-        As imputation logic for numerical features can be quite intricate,
-        it is left to the user to manage.
-        See also attribute :attr:`~skrub.TableVectorizer.imputed_columns_`.
+    activate_auto_cast : bool, default=True
+        If set to ``True``, calling ``fit``, ``transform`` or ``fit_transform``
+        will call ``auto_cast`` to convert each column to the "optimal" dtype
+        for scikit-learn estimators.
+        The main heuristics are the following:
+        - pandas extension dtypes conversion to numpy dtype
+        - datetime conversion using ``skrub.to_datetime``
+        - numeric conversion using ``pandas.to_numeric``
+        - numeric columns with missing values are converted to float to input np.nan
+        - categorical columns dtypes are updated with the new entries (if any)
+          during transform.
 
     remainder : {'drop', 'passthrough'} or Transformer, default='passthrough'
         By default, all remaining columns that were not specified in `transformers`
         will be automatically passed through. This subset of columns is concatenated
         with the output of the transformers. (default 'passthrough').
-        By specifying `remainder='drop'`, only the specified columns
+        By specifying ``remainder='drop'``, only the specified columns
         in `transformers` are transformed and combined in the output, and the
         non-specified columns are dropped.
         By setting `remainder` to be an estimator, the remaining
         non-specified columns will use the `remainder` estimator. The
-        estimator must support :term:`fit` and :term:`transform`.
+        estimator must support ``fit`` and ``transform``.
         Note that using this feature requires that the DataFrame columns
-        input at :term:`fit` and :term:`transform` have identical order.
+        input at ``fit`` and ``transform`` have identical order.
 
     sparse_threshold : float, default=0.0
         If the output of the different transformers contains sparse matrices,
         these will be stacked as a sparse matrix if the overall density is
-        lower than this value. Use `sparse_threshold=0` to always return dense.
+        lower than this value. Use ``sparse_threshold=0`` to always return dense.
```

Comment:
here we should probably note that with the default encoders output will never be sparse